go/src/koding/db/models/error.go

```diff
@@ -0,0 +1,65 @@
+package models
+
+import (
+	"fmt"
+	"strings"
+
+	mgo "gopkg.in/mgo.v2"
+)
+
+// NotFoundError is returned when requested operation on a resource
+// cannot be completed due to resource being missing. Most likely
+// this kind of errors means the resource was deleted outside
+// (shared ownership) and this error is returned to the caller,
+// so it recover if possible.
+//
+// TODO(rjeczalik): the db/mongo helpers should be decorating errors
+// instead, so database handling layer is transparent and no
+// library specific errors (e.g. mgo.ErrNotFound, pq.Error) are leaked.
```

Comment:
I've removed the comment - it was about making modelhelper package, not models one, do not return mgo.* types and values.

The idea behind this one was to decouple mongo / postgres from packages that require database access. For new package I try to achieve that by creating interfaces for database access, like:

- [credential.Store](https://github.com/koding/koding/blob/55dbb189af66bb1d5b6410bfc8568721e826b79b/go/src/koding/kites/kloud/credential/store.go#L67-L71)
- [credential.Database](https://github.com/koding/koding/blob/55dbb189af66bb1d5b6410bfc8568721e826b79b/go/src/koding/kites/kloud/credential/client.go#L60-L103)
- [provider.Database](https://github.com/koding/koding/blob/55dbb189af66bb1d5b6410bfc8568721e826b79b/go/src/koding/kites/kloud/stack/provider/database.go#L83-L108)

In the long run my aim is to make it possible to have tests for stack template building which do not require actual database and fixtures.